src/FxCop/System.Runtime.Analyzers/VisualBasic/Design/TypesThatOwnDisposableFieldsShouldBeDisposable.vb

```diff
@@ -0,0 +1,51 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace System.Runtime.Analyzers
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Public Class BasicTypesThatOwnDisposableFieldsShouldBeDisposableAnalyzer
+        Inherits TypesThatOwnDisposableFieldsShouldBeDisposableAnalyzer(Of TypeBlockSyntax)
+        Protected Overrides Function GetAnalyzer(disposableTypeSymbol As INamedTypeSymbol) As DisposableFieldAnalyzer
+            Return New BasicDisposableFieldAnalyzer(disposableTypeSymbol)
+        End Function
+
+        Private Class BasicDisposableFieldAnalyzer
+            Inherits DisposableFieldAnalyzer
+            Public Sub New(disposableTypeSymbol As INamedTypeSymbol)
+                MyBase.New(disposableTypeSymbol)
+            End Sub
+
+            Protected Overrides Function IsDisposableFieldCreation(node As SyntaxNode, model As SemanticModel, disposableFields As HashSet(Of ISymbol), cancellationToken As CancellationToken) As Boolean
+                If TypeOf node Is AssignmentStatementSyntax Then
+                    Dim assignment = DirectCast(node, AssignmentStatementSyntax)
+                    If TypeOf assignment.Right Is ObjectCreationExpressionSyntax AndAlso disposableFields.Contains(model.GetSymbolInfo(assignment.Left, cancellationToken).Symbol) Then
+                        Return True
+                    End If
+                ElseIf TypeOf node Is FieldDeclarationSyntax Then
+                    Dim fieldDecls = DirectCast(node, FieldDeclarationSyntax).Declarators
+                    For Each declarator As VariableDeclaratorSyntax In fieldDecls
+                        If declarator.Names.Count > 1 Then
+                            Continue For
+                        End If
+                        Dim fieldName = declarator.Names.First()
+                        If TypeOf declarator?.Initializer?.Value Is ObjectCreationExpressionSyntax AndAlso disposableFields.Contains(model.GetDeclaredSymbol(fieldName, cancellationToken)) Then
+                            Return True
+                        End If
+                    Next
```

Comment:
@tmeschter Actually I think VB check is fine, since the code above dealt with the case where one declaration statement has multiple declarators. For Example, following code will trigger the warning:
```VB
    Public Class NoDisposeClass
        Dim newFile1 As FileStream, newFile2 As FileStream = New FileStream(""data.txt"", FileMode.Append)
    End Class
```
Unlike C#, explicit initialization is not permitted with multiple variables declared with a single type specifier in VB, so the check above on whether `declarator.Names.Count > 1` is for this case. I'm learning VB while implementing those analyzers, so please let me know if I was wrong :)